clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,921 @@
+		U(c.Username).
+		P(c.Password).
+		Db(c.Database).
+		Q(query).
+		Rp(c.RetentionPolicy).
+		Accept(resContentType).
+		Execute()
+	if err != nil {
+		return "", err
+	}
+	return resBody, nil
+}
+
+func (c *Client) setFormat(args []string) {
+	// args[0] is "format"
+	if len(args) != 2 {
+		color.Red("Expected a format [csv, json, table]")
+		return
+	}
+	newFormat := FormatType(args[1])
+	switch newFormat {
+	case CsvFormat, JsonFormat, TableFormat:
+		c.Format = newFormat
+	default:
+		color.HiRed("Unimplemented format %q, keeping %s format.", newFormat, c.Format)
+		color.HiBlack("Choose a format from [csv, json, table]")
+	}
+}
+
+func (c *Client) setPrecision(args []string) {
+	// args[0] is "precision"
+	if len(args) != 2 {
+		color.Red("Expected a precision [ns, u, ms, s, m, or h]")
+		return
+	}
+	precision := args[1]
+	switch precision {
+	case "ns", "u", "µ", "ms", "s", "m", "h":
+		c.Precision = precision
+	default:
+		color.HiRed("Unimplemented precision %q, keeping %s precision.", precision, c.Precision)
+		color.HiBlack("Choose a precision from [ns, u, ms, s, m, or h]")
+	}
+}
+
+func (c *Client) outputCsv(csvBody string) {
+	fmt.Println(csvBody)
+}
+
+func (c *Client) outputJson(jsonBody string) {
+	if !c.Pretty {
+		fmt.Println(jsonBody)
+	} else {
+		var buf bytes.Buffer
+		if err := json.Indent(&buf, []byte(jsonBody), "", "  "); err != nil {
+			color.Red("Unable to prettify json response.")
+			fmt.Println(jsonBody)
+		} else {
+			fmt.Println(buf.String())
+		}
+	}
+}
+
+func (c *Client) outputTable(jsonBody string) {
+	var responses api.InfluxqlJsonResponse
+	if err := json.Unmarshal([]byte(jsonBody), &responses); err != nil {
+		color.Red("Failed to parse JSON response")
+	}
+	for _, res := range responses.GetResults() {
+		for _, series := range res.GetSeries() {
+			color.Magenta("Table View (press q to exit interactive mode):")
+			p := tea.NewProgram(NewModel(series))
+			if err := p.Start(); err != nil {
+				color.Red("Failed to display table")
+			}
+			fmt.Printf("\n")
+		}
+	}
+}
+
+func (c *Client) togglePretty() {
+	c.Pretty = !c.Pretty
+	color.HiBlack("Pretty: %v", c.Pretty)
+}
+
+func (c *Client) use(args []string) {
+	if len(args) != 2 {
+		color.Red("wrong number of args for \"use <database>\"")
+		return
+	}
+	parsedDb, parsedRp, err := parseDatabaseAndRetentionPolicy([]byte(args[1]))
+	if err != nil {
+		color.Red("Unable to parse: %v", err)
+		return
+	}
+	dbs, err := c.GetDatabases(context.Background())
+	if err != nil {
+		color.Red("Unable to check databases: %v", err)
+		return
+	}
+	// discover if the parsedDb is a valid database
+	for _, db := range dbs {
+		if parsedDb == db {
+			exists := false
+			prevDb := c.Database
+			c.Database = parsedDb
+			rps, _ := c.getRetentionPolicies(context.Background())
+			// discover if the parsedRp is a valid retention policy
+			for _, rp := range rps {
+				if parsedRp == rp || parsedRp == "" {
+					if parsedRp == "" {
```

Comment:
We can hoist the `parsedRp == ""` check out of the loop

Reply:
Moved it to a switch statement which `continue`s in the`default` case. Much cleaner imo.